### PR TITLE
strongly connected components using Tarjan's algorithm in c++

### PR DESCRIPTION
**Pull request by TVSSSRIPAD**
Fixes issue #2073

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- added the code for finding strongly connected components using tarjan's algorithm in c++ 

#### Languages Used:
- c++ 

#### Files Added:
- tarjan_scc.cpp 

Thanks!

### CODE REVIEW

File: Tarjan_strongly_connected_components/tarjan_scc.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
+#include<bits/stdc++.h>
+int NIL = -1;
+using namespace std; 
+
+void addEdge(list<int> *adj,int v, int w)
```
Comment (asha15) [inline Tarjan_strongly_connected_components/tarjan_scc.cpp:5]: keep pace after comma

File: Tarjan_strongly_connected_components/tarjan_scc.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
+#include<bits/stdc++.h>
+int NIL = -1;
+using namespace std; 
+
+void addEdge(list<int> *adj,int v, int w) 
+{ 
+    adj[v].push_back(w); 
+} 
+
+void DFS(int V, list<int>*adj,int u, int dis[], int low[], stack<int> *s, bool visited[])
```
Comment (asha15) [inline Tarjan_strongly_connected_components/tarjan_scc.cpp:10]: keep a space after comma

File: Tarjan_strongly_connected_components/tarjan_scc.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+void addEdge(list<int> *adj,int v, int w) 
+{ 
+    adj[v].push_back(w); 
+} 
+
+void DFS(int V, list<int>*adj,int u, int dis[], int low[], stack<int> *s, bool visited[]) 
+{ 
+    static int time = 0; 
+	//updating the low value , discovery time of vertex
```
Comment (asha15) [inline Tarjan_strongly_connected_components/tarjan_scc.cpp:13]: correct the indentation

File: Tarjan_strongly_connected_components/tarjan_scc.cpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    s->push(u); 
+    visited[u] = true; 
+
+    list<int>::iterator i; 
+    for (i = adj[u].begin(); i != adj[u].end(); ++i) 
+    { 
+        int v = *i;  
+        if (dis[v] == -1) 
+        { 
+            DFS(V,adj,v, dis, low, s, visited);
```
Comment (asha15) [inline Tarjan_strongly_connected_components/tarjan_scc.cpp:24]: keep a space after comma

File: Tarjan_strongly_connected_components/tarjan_scc.cpp [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        if (dis[v] == -1) 
+        { 
+            DFS(V,adj,v, dis, low, s, visited); 
+            low[u]  = min(low[u], low[v]); 
+        } 
+ 
+        else if (visited[v] == true) 
+            low[u]  = min(low[u], dis[v]); 
+    } 
+
```
Comment (asha15) [inline Tarjan_strongly_connected_components/tarjan_scc.cpp:31]: remove this extra line

File: Tarjan_strongly_connected_components/tarjan_scc.cpp [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        while (s->top() != u) 
+        { 
+            w = (int) s->top(); 
+            cout << w << " ";          
+            s->pop(); 
+        } 
+        w = (int) s->top(); 
+        cout << w << "\n";        
+        s->pop(); 
+    }
```
Comment (asha15) [inline Tarjan_strongly_connected_components/tarjan_scc.cpp:45]: correct the indentation

File: Tarjan_strongly_connected_components/tarjan_scc.cpp [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    	int w = 0;  // To store the extracted vertices 
+        while (s->top() != u) 
+        { 
+            w = (int) s->top(); 
+            cout << w << " ";          
+            s->pop(); 
+        } 
+        w = (int) s->top(); 
+        cout << w << "\n";        
+        s->pop();
```
Comment (asha15) [inline Tarjan_strongly_connected_components/tarjan_scc.cpp:44]: keep a space after ->

File: Tarjan_strongly_connected_components/tarjan_scc.cpp [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+	for(int i=0;i<E;i++)
+	{
+		cin>>u>>v;
+		addEdge(adj,u,v); //adding edges to graph
+	}	
+    
+	//printing the adjacency list to check if it is created correctly
+	cout<<"Adjacency list : \n";
+    for(auto i=0;i<V;i++)
+    {
```
Comment (asha15) [inline Tarjan_strongly_connected_components/tarjan_scc.cpp:69]: correct the indentation 
keep a space after operator

File: Tarjan_strongly_connected_components/tarjan_scc.cpp [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+    // Initializing dis and low, and visited arrays 
+    for (int i = 0; i < V; i++) 
+    { 
+        dis[i] = NIL; 
+        low[i] = NIL; 
+        visited[i] = false; 
+    } 
+  
+    // Call the DFS function to find strongly connected components in DFS tree with vertex 'i' 
+    cout<<"The Strongly connected Components are : \n";
```
Comment (asha15) [inline Tarjan_strongly_connected_components/tarjan_scc.cpp:89]: correct the indentation
keep a space after operator

File: Tarjan_strongly_connected_components/tarjan_scc.cpp [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+    addEdge(adj,0,3); 
+    addEdge(adj,3, 4);
+    The Strongly Connected Components are : 
+    4
+    3
+    1 2 0  
+    */ 
+
+    return 0; 
+}
```
Comment (asha15) [inline Tarjan_strongly_connected_components/tarjan_scc.cpp:109]: correct the indentation
keep a space after operator

File: Tarjan_strongly_connected_components/tarjan_scc.cpp [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+    addEdge(adj,1, 0); 
+    addEdge(adj,0,2); 
+    addEdge(adj,2,1); 
+    addEdge(adj,0,3); 
+    addEdge(adj,3, 4);
+    The Strongly Connected Components are : 
+    4
+    3
+    1 2 0  
+    */
```
Comment (asha15) [inline Tarjan_strongly_connected_components/tarjan_scc.cpp:106]: Take this to the bottom of the file

**Comment by raghav-dalmia** [comment]
Make sure to squash your commits.

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by somya-kapoor** [comment]
Due to no activity for more than 7 days, I am closing this issue.

**Pull request closed without merging**